src/utils/date.ts

```diff
@@ -10,3 +12,63 @@ export const formatDate = (date: string | Date, formatStr = DEFAULT_DATE_FORMAT)
     throw new Error(error.message);
   }
 };
+
+type Asset = {
+  datetime?: string;
+  created?: string;
+  updated?: string;
+  start_datetime?: string;
+  end_datetime?: string;
+};
+
+type StacData = Collection | StacItem;
+
+export type ExtractedDates = {
+  start: string | null;
+  end: string | null;
+  datetime: string | null;
+  created: string | null;
+  updated: string | null;
+  start_datetime: string | null;
+  end_datetime: string | null;
+};
+
+export const extractDates = (stacData: StacData): ExtractedDates => {
+  const dates: ExtractedDates = {
+    start: null,
+    end: null,
+    datetime: null,
+    created: null,
+    updated: null,
+    start_datetime: null,
+    end_datetime: null,
+  };
+
+  // Handle Collection temporal extent
+  if (stacData.type === 'Collection') {
+    const temporalExtent = stacData?.extent?.temporal?.interval?.[0]; // First interval
+    if (temporalExtent) {
+      dates.start = temporalExtent[0] || null;
+      dates.end = temporalExtent[1] || null;
+    }
+  }
+
+  // Handle Item datetime in properties
+  if (stacData.type === 'Feature') {
+    dates.start = stacData.properties.datetime || stacData.properties.start_datetime || null;
+    dates.end = stacData.properties.end_datetime || null;
```

Comment:
I'm not sure about the priority order of dates here.

If an item or dataset has a range of dates then we should display and use the range as this is the most useful. eg, Sentinel 2 has data from 2015 to 2024 so we should show that range even if a single timestamp of, say, 2019 has also been provided. The STAC standard (https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#datetime-selection) recommends populating both the range and the single date, with the single date being a 'nominal or representative datetime'.

I see a couple of problems here:
* If `datetime`, `start_datetime` and `end_datetime` are all set then we'll use the `datetime` for the start and `end_datetime` for the end. But this means reporting 2019-2024 in the Sentinel 2 example which not a range from anywhere in the record and would be an incorrect statement. If the range is set we should use the range, if not we should use a single date, and never mix the two.
* This can't distinguish between a range and a point. A `start` in 2015 and an `end` of `null` should mean 2015 onwards. However, if `datetime` is set and `{start,end}_datetime` are not then we should be displaying it as a point in time, not a range. `datetime` in 2015 and the range `null` would result in '2015 onwards' being displayed as well, which is wrong.

Reply:
I improved the getStacDate in stacUtils.tsx to prioritise ranges over using datetime. In addition to this it can handle if the start or end of a range is missing